parquet-hadoop/src/test/java/org/apache/parquet/statistics/TestFloat16Statistics.java

```diff
@@ -0,0 +1,157 @@
+
+import org.apache.parquet.Preconditions;
+import org.apache.parquet.example.data.Group;
+import org.apache.parquet.example.data.GroupFactory;
+import org.apache.parquet.example.data.simple.SimpleGroupFactory;
+import org.apache.parquet.hadoop.ParquetFileReader;
+import org.apache.parquet.hadoop.ParquetWriter;
+import org.apache.parquet.internal.column.columnindex.ColumnIndex;
+import org.apache.parquet.io.api.Binary;
+import org.apache.parquet.schema.MessageType;
+import org.apache.parquet.schema.Types;
+import org.apache.parquet.type.Float16;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.parquet.column.statistics.Statistics;
+import org.apache.parquet.hadoop.example.ExampleParquetWriter;
+import org.apache.parquet.hadoop.example.GroupWriteSupport;
+import org.apache.parquet.hadoop.metadata.ColumnChunkMetaData;
+import org.apache.parquet.hadoop.util.HadoopInputFile;
+
+import static org.apache.parquet.schema.LogicalTypeAnnotation.float16Type;
+import static org.apache.parquet.schema.PrimitiveType.PrimitiveTypeName.FIXED_LEN_BYTE_ARRAY;
+import static org.junit.Assert.assertEquals;
+
+public class TestFloat16Statistics {
+
+  @Rule
+  public TemporaryFolder temp = new TemporaryFolder();
+
+  private short[] valuesInAscendingOrder = {
+    (short) 0xfc00, // -Infinity
+    (short) 0xc000, // -2.0
+    -Float16.MAX_VALUE, // -6.109476E-5
+    Float16.NEGATIVE_ZERO, // -0
+    Float16.POSITIVE_ZERO, // +0
+    Float16.MIN_VALUE, // 5.9604645E-8
+    Float16.MAX_VALUE, // 65504.0
+    (short) 0x7c00}; // Infinity
+
+  @Test
+  public void testFloat16ColumnIndex() throws IOException
+  {
+    MessageType schema = Types.buildMessage().
+      required(FIXED_LEN_BYTE_ARRAY).as(float16Type()).length(2).named("col_float16").named("msg");
+
+    Configuration conf = new Configuration();
+    GroupWriteSupport.setSchema(schema, conf);
+
+    GroupFactory factory = new SimpleGroupFactory(schema);
+    Path path = newTempPath();
+    try (ParquetWriter<Group> writer = ExampleParquetWriter.builder(path)
+      .withConf(conf)
+      .withDictionaryEncoding(false)
+      .build()) {
+
+      for (short value : valuesInAscendingOrder) {
+        writer.write(factory.newGroup().append("col_float16", Binary.fromConstantByteArray(Float16.toBytesLittleEndian(value))));
+      }
+    }
+
+    try (ParquetFileReader reader = ParquetFileReader.open(HadoopInputFile.fromPath(path, new Configuration()))) {
+
+      ColumnChunkMetaData column = reader.getFooter().getBlocks().get(0).getColumns().get(0);
+      ColumnIndex index = reader.readColumnIndex(column);
+      assertEquals(Collections.singletonList((short) 0xfc00), toFloat16List(index.getMinValues()));
```

Comment:
It would be nice if we can verify the different boundary order of column index (like ascending, descending, etc.). To achieve this, we might need more values.

Reply:
We also need to test these cases:
- NaN values are present.
- All values are +0 or -0.
- Values are in ascending/descending/undefined order.

Reply:
good suggestions, added more tests